kernel/src/actions/mod.rs

```diff
@@ -152,6 +199,81 @@ impl Protocol {
             .as_ref()
             .is_some_and(|features| features.iter().any(|f| f == feature.as_ref()))
     }
+
+    /// Check if reading a table with this protocol is supported. That is: does the kernel support
+    /// the specified protocol reader version and all enabled reader features? If yes, returns unit
+    /// type, otherwise will return an error.
+    pub fn check_read_supported(&self) -> DeltaResult<()> {
+        match &self.reader_features {
+            // if min_reader_version = 3 and all reader features are subset of supported => OK
+            Some(reader_features) if self.min_reader_version == 3 => {
+                check_supported_features(reader_features, &SUPPORTED_READER_FEATURES)
+            }
+            // if min_reader_version = 3 and no reader features => ERROR
+            // note this is caught by the protocol parsing.
+            None if self.min_reader_version == 3 => unreachable!(),
+            // if min_reader_version = 1,2 and there are no reader features => OK
+            None if self.min_reader_version == 1 || self.min_reader_version == 2 => Ok(()),
+            // if min_reader_version = 1,2 and there are reader features => ERROR
+            // note this is caught by the protocol parsing.
+            Some(_) if self.min_reader_version == 1 || self.min_reader_version == 2 => {
```

Comment:
What are we trying to catch here? `unreachable!` will raise some odd panic that probably won't help debugging. Maybe return an `InternalError`?

Reply:
ah nice yea meant to follow up on this. I think I'll do `InternalError` and just note in those comments that it's caught by parsing and we shouldn't hit that. didn't know if we wanted to do any debug assert or something. let me know if you have any better ideas. just became a funky spot since it's prohibited by the parsing but then it feels like it's 'left out' if it's not included. an obvious alternative is just deleting these two cases (caught by parsing) entirely